docs/architecture-decision-log/README.md

```diff
@@ -0,0 +1,146 @@
+# Architecture Decision Log
+
+Welcome in the Tournesol Architecture Decision Log (A.D.L.).
+
+This documentation contains all important architectural decisions, and
+software design choices made during the development of the Tournesol platform.
+
+All decisions are captured within individual files, called Architecture
+Decision Record (A.D.R.).
+
+Learn more about A.D.R. in [this other documentation][github-adr].
+
+---
+
+Table of Content
+
+* [Keywords](#keywords)
+* [Usage](#summary)
+  * [Rules](#rules)
+  * [Add a new decision record](#issue)
+  * [Update a decision record's status](#decision)
+* [Documentation](#documentation)
+  * [The A.D.R. template](#the-ADR-template)
+  * [Examples](#examples)
+* [License](#license)
+
+## Keywords
+
+**A.D.** - architecture decision
+
+An architecture decision (A.D.) is a software design choice that addresses a
+significant requirement.
+
+**A.D.R.** - architecture decision record
+
+An architecture decision record (A.D.R.) is a document that captures an important
+architectural decision made along with its context and consequences.
+
+**A.D.L.** - architecture decision log
+
+An architecture decision log (A.D.L.) is the collection of all ADRs created and
+maintained for a particular project or organization.
+
+**AKM** - architecture knowledge management
+
+All these are within the topic of architecture knowledge management (A.K.M.).
+
+## Usage
+
+### Rules
+
+Characteristics of a good A.D.R.
+
+- **Point in Time** - Identify when the A.D. was made
+- **Rationality** - Explain the reason for making the particular A.D.
+- **Immutable record** - The decisions made in a previously published A.D.R.
+  should not be altered
+- **Specificity** - Each A.D.R. should be about a single A.D.
+
+A new A.D.R. may take the place of a previous A.D.R.
+
+When an A.D. is made that replaces or invalidates existing A.D.R.
+- a new A.D.R. should be created ;
+- and the status of the old ones should be updated.
+
+### Add a new decision record
+
+First, copy the [A.D.R. template](./adr_template.md) in the appropriate
+sub-folder.
+
+If this new decision is related to a new topic, create a new sub-folder
+instead.
+
+Prefix the newly created file by the date of the decision, following the
+format `YYYY_MM_DD_`.
```

Comment:
A small drawback of this approach where each A.D.R is nested under a sub-folder is that it's not obvious to find the latest decisions, without using the git history or fetching the repo. 

Nevertheless I agree that grouping the decisions by topic is useful. So I don't have any better suggestion :slightly_smiling_face:

Reply:
That's true.

Each decision are grouped by topic, so that it's easier to retrieve them by browsing the A.D.L.

Is that possible we want to retrieve the latest decision we made without knowing the related topic? I hope this situation won't occur often : )

If it's the case, we can add an history section in the A.D.L. `README.md`, containing the 4 latest decision for instance. Or, we can even add a complete `CHANGELOG.md` so that we can track each events in the chronological order.

Example of a simple `CHANGELOG.md`:

```
# Changelog

## [future]

## 2021-12

- approved   2021-12-12 Create an ADL
- rejected   2021-09-04 something

## 2021-09

- proposed     2021-09-20 something
- superseded   2021-09-01 do that by 2021-09-02 do this instead
- approved     2021-09-02 do this instead
- approved     2021-09-01 do that
```

Reply:
Considering the following:

- maintaining this kind of history is simple ( add one more line after adding / editing a decision )
- we don't take too many decisions by month right now
- we have no process to notify everyone when a new decision is taken

I propose to add a `CHANGELOG.md` in the A.D.L. It will also help people re-joining the project after a long break.